feat(MA): Adding row and cell actions

jabahamondes opened the pull request:
Note: a lot snapshots changed, but there shouldn't be any actual UI visible change

## Problem

I have the need to add actions for rows and cells

## Changes

Add the capability to define actions for rows and cells. For this I added a cell actions wrapper that include the `More` icon  + the action `cellActionsRenderer`. Also including similar logic in the data table for the row actions. Added a storybook to show how the actions will be coded. 

Note: There is already an action for the activity table, but it's quite hardcoded. In the future we could refactor it and use the new custom actions for the row.


## How did you test this code?

Manually and with storybook:

<img width="440" height="411" alt="Screenshot 2025-11-27 at 11 06 14 AM" src="https://github.com/user-attachments/assets/4d566284-edb9-483c-b8bf-b49e1de70468" />

<img width="388" height="397" alt="Screenshot 2025-11-27 at 11 06 11 AM" src="https://github.com/user-attachments/assets/275b68f2-e863-4256-9109-d547e510d1a3" />

This is how it looks like the last cell hovered with action + a row action
<img width="1272" height="421" alt="Screenshot 2025-11-27 at 11 06 08 AM" src="https://github.com/user-attachments/assets/3d5cc8c5-9dd9-46ab-8c9f-f998511b0ba9" />

jabahamondes committed:
Adding row and cell actions

github-actions[bot] committed:
test(storybook): update UI snapshots

jabahamondes committed:
Removing unnecessary change from another pr

jabahamondes committed:
Merge branch 'feat/adding-row-and-cell-actions-3' of github.com:PostHog/posthog into feat/adding-row-and-cell-actions-3

github-actions[bot] committed:
test(storybook): update UI snapshots

jabahamondes committed:
Improving comments

jabahamondes committed:
Merge branch 'feat/adding-row-and-cell-actions-3' of github.com:PostHog/posthog into feat/adding-row-and-cell-actions-3

github-actions[bot] committed:
test(storybook): update UI snapshots

thmsobrmlr reviewed (commented):
Cool work! My main concern is that this changes the paradigm of the table. In your version the rows and columns are passed in from the outside, which turns the table into a regular "dumb component". In its current form the table has inherent knowledge of the query and uses `QueryFeatures` to decide what to render. I’m unsure if we should combine these two approaches. Maybe @mariusandra can weigh in as the original author.

---

From a code perspective everything looks good. Thanks for adding the storybook stories as well @jabahamondes!

Some small comments:
- The cell actions are visually hard to discover. Even knowing they were there I still struggled to see them. Maybe mark this in the header column, adjust the cursor on the cell or highlight the row on hover.
- We now end up with two implementations of row actions, this one and EventRowActions. as you already suggest, it'd be great to end up with only one implementation eventually.

jabahamondes committed:
Always displaying the actions if the cell have them to be easier to for them to be discovered

jabahamondes committed:
Merge remote-tracking branch 'origin/master' into feat/adding-row-and-cell-actions-3

github-actions[bot] committed:
test(storybook): update UI snapshots

jabahamondes committed:
feat(table): add cell and row action support in LemonTable

- Introduced cellActions for individual cell actions in table columns.
- Added rowActions for actions at the end of each row.
- Updated LemonTable and TableRow components to handle new action props.
- Enhanced storybook examples to demonstrate new action functionalities.

This improves user interaction by allowing specific actions directly from table cells and rows, making it easier to manage data entries. -> PRs comments

github-actions[bot] committed:
test(storybook): update UI snapshots

thmsobrmlr reviewed (approved):
Awesome! I like it much more with the always visible cell actions.

jabahamondes committed:
Merge remote-tracking branch 'origin/master' into feat/adding-row-and-cell-actions-3

jabahamondes committed:
Merge branch 'feat/adding-row-and-cell-actions-3' of github.com:PostHog/posthog into feat/adding-row-and-cell-actions-3
